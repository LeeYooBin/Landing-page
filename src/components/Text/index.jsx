import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Content = styled.p`
  font: ${({ theme }) => theme.fonts.text};
  color: ${({ theme }) => theme.colors.text};
  opacity: 0.7;
`;

const Text = ({ text }) => <Content>{text}</Content>;

Text.propTypes = {
  text: PropTypes.string,
};

export default Text;
