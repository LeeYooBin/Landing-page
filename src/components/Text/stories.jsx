import Text from '.';

export default {
  title: 'Text',
  component: Text,
};

export const Template = (args) => <Text {...args} />;
