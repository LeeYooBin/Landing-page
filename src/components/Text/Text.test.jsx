import { describe, it, expect } from '@jest/globals';
import { screen } from '@testing-library/react';
import Text from '.';
import renderTheme from '../../styles/render-theme';

describe('Text component', () => {
  it('Should render with default values', () => {
    renderTheme(<Text />);
  });

  it('Should render the given text without errors', () => {
    renderTheme(<Text text="Text" />);
    expect(screen.getByText('Text')).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<Text text="Text" />);
    expect(container).toMatchSnapshot();
  });

  it('verifies propTypes', () => {
    const consoleError = console.error;
    // eslint-disable-next-line no-undef
    console.error = jest.fn();

    renderTheme(<Text text={123} />);
    expect(console.error).toHaveBeenCalled();
    expect(console.error).toHaveBeenCalledTimes(1);

    console.error.mockRestore();
    console.error = consoleError;
  });
});
