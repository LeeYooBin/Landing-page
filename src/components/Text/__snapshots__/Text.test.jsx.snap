// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Should match snapshot 1`] = `
.c0 {
  font: 500 1.4rem/2rem "Heebo",sans-serif;
  color: #FFF;
  opacity: 0.7;
}

<div>
  <p
    class="c0"
  >
    Text
  </p>
</div>
`;
